firefox-ios/Client/Frontend/PasswordGenerator/PasswordGeneratorViewController.swift

```diff
@@ -0,0 +1,293 @@
+                constant: UX.passwordLabelAndButtonSpacing),
+            passwordRefreshButton.trailingAnchor.constraint(
+                equalTo: passwordField.trailingAnchor,
+                constant: -UX.passwordFieldHorizontalPadding),
+            passwordRefreshButton.centerYAnchor.constraint(
+                equalTo: passwordField.centerYAnchor),
+            passwordRefreshButton.widthAnchor.constraint(equalToConstant: scaledRefreshButtonSize),
+            passwordRefreshButton.heightAnchor.constraint(equalToConstant: scaledRefreshButtonSize),
+        ])
+    }
+
+    // MARK: - Themable
+    func applyTheme() {
+        let theme = themeManager.getCurrentTheme(for: windowUUID)
+        view.backgroundColor = theme.colors.layer1
+        headerImageView.tintColor = theme.colors.iconPrimary
+        headerLabel.textColor = theme.colors.textPrimary
+        descriptionLabel.textColor = theme.colors.textSecondary
+        passwordField.backgroundColor = theme.colors.layer2
+        passwordField.layer.borderColor = theme.colors.borderPrimary.cgColor
+        passwordLabel.textColor = theme.colors.textPrimary
+        passwordRefreshButton.tintColor = theme.colors.iconPrimary
+        usePasswordButton.applyTheme(theme: theme)
+    }
+
+    private func buildPasswordLabel() {
+        passwordLabel.numberOfLines = 0
+        passwordLabel.font = FXFontStyles.Regular.body.scaledFont()
+        passwordLabel.text = passwordGeneratorState.password
+        passwordLabel.accessibilityAttributedLabel = generateAccessibilityAttributedLabel()
+    }
+
+    private func buildUsePasswordButton() {
+        let usePasswordButtonVM = PrimaryRoundedButtonViewModel(
+            title: .PasswordGenerator.UsePasswordButtonLabel,
+            a11yIdentifier: AccessibilityIdentifiers.PasswordGenerator.usePasswordButton)
+        usePasswordButton.configure(viewModel: usePasswordButtonVM)
+    }
+
+    // MARK: - Redux
+    func subscribeToRedux() {
+        store.dispatch(
+            ScreenAction(
+                windowUUID: windowUUID,
+                actionType: ScreenActionType.showScreen,
+                screen: .passwordGenerator
+            )
+        )
+
+        let uuid = windowUUID
+        store.subscribe(self, transform: {
+            return $0.select({ appState in
+                return PasswordGeneratorState(appState: appState, uuid: uuid)
+            })
+        })
+    }
+
+    func unsubscribeFromRedux() {
+        store.dispatch(
+            ScreenAction(
+                windowUUID: windowUUID,
+                actionType: ScreenActionType.closeScreen,
+                screen: .passwordGenerator
+            )
+        )
+    }
+
+    func newState(state: PasswordGeneratorState) {
+        passwordGeneratorState = state
+        passwordLabel.text = passwordGeneratorState.password
+        passwordLabel.accessibilityAttributedLabel = generateAccessibilityAttributedLabel()
```

Comment:
Just a note to keep in mind - If this gets called multiple times, so might want to make sure we update these as needed when those state details change.

Reply:
Good call! The plan right now is that the only thing triggering a state change for the password generator is changing the password, so each time the state changes, these two would need to be updated